airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -401,15 +402,29 @@ def write_dag(
         # If Yes, does nothing
         # If No or the DAG does not exists, updates / writes Serialized DAG to DB
         if min_update_interval is not None:
-            if session.scalar(
-                select(literal(True))
-                .where(
-                    cls.dag_id == dag.dag_id,
-                    (timezone.utcnow() - timedelta(seconds=min_update_interval)) < cls.last_updated,
+            try:
+                do_nothing = session.scalar(
+                    select(literal(True))
+                    .where(
+                        cls.dag_id == dag.dag_id,
+                        (timezone.utcnow() - timedelta(seconds=min_update_interval)) < cls.last_updated,
+                    )
+                    .select_from(cls)
+                    .with_for_update(nowait=True, skip_locked=True)
                 )
-                .select_from(cls)
-            ):
-                return False
+                if do_nothing:
+                    return False
+            except OperationalError as e:
+                if (
+                    hasattr(e, "orig") and hasattr(e.orig, "pgcode") and e.orig.pgcode == "55P03"
```

Comment:
I thinbk there's already a helper method for this -- we support more than just Postgres databases